pkg/client/cli/output/output.go

```diff
@@ -132,23 +141,51 @@ func Execute(cmd *cobra.Command) (*cobra.Command, bool, error) {
 		}
 		obj = response
 	}
-	switch o.format {
+
+	o.write(obj)
+	return cmd, true, err
+}
+
+func shapeObjInfo(cmd *cobra.Command, obj any) *object {
+	response := &object{
+		Cmd: cmd.Name(),
+	}
+
+	response.Stdout = obj
+
+	if buf, ok := cmd.ErrOrStderr().(*bytes.Buffer); ok && buf.Len() > 0 {
+		response.Stderr = buf.String()
+	}
+
+	return response
+}
+
+// makeWriterForFormat creates an appropriate writer for the given format.
+func makeWriterForFormat(format format, originalStdout io.Writer) func(obj any) {
+	switch format {
 	case formatJSON:
-		if encErr := json.NewEncoder(o.originalStdout).Encode(obj); encErr != nil {
-			panic(encErr)
+		// create the encoder once
+		encoder := json.NewEncoder(originalStdout)
+		return func(obj any) {
+			if err := encoder.Encode(obj); err != nil {
+				panic(err)
+			}
 		}
```

Comment:
This will produce invalid JSON on the encoder's output when the encoder is called more than once. It'll just print the objects, E.g.:
```
{"X":"x"}
{"Y":"y"}
```
The correct JSON must be a comma separated array:
```
[{"X":"x"},{"Y":"y"}]
```
And this is the reason why streaming is difficult.

Reply:
I should have provided more context. The problem comes from the display of services in the docker extension. If the user adds a service to the cluster, it should also appear in the table of intercept-able services shown by the extension. To do this, we call `telepresence list --output json --watch`, which I wrote in the past to print a new list object whenever the list of services changed. Whenever a new object is printed, the extension would read it in and update the table accordingly. 

`{"X":"x"}
{"Y":"y"}
...`

is what I want.

Reply:
I understand. So how about this:
1. Add a new output format called "json-stream", which we document as having your desired characteristics.
2. Keep the original code as is, except ines 65-72, which, when that format is used, instead does:
```
		if encErr := json.NewEncoder(o.originalStdout).Encode(obj); encErr != nil {
			panic(encErr)
		}
		o.override  = true
```
Also, add a `case formatJSON_STREAM:` to the switch at line 136 that does nothing.

The `--watch` flag should not be combinable with `--format json`. I think `--format yaml` would be OK though, because the output is still valid yaml.